testing/reset_error.cu

```diff
@@ -0,0 +1,29 @@
+#include <nvbench/cuda_call.cuh>
+
+#include "test_asserts.cuh"
+
+namespace
+{
+    __global__ void multiply5(const int32_t* __restrict__ a, int32_t* __restrict__ b)
+    {
+      const auto id = blockIdx.x * blockDim.x + threadIdx.x;
+      b[id] = 5 * a[id];
+    }
+}
+
+int main()
+{ 
+  multiply5<<<256, 256>>>(nullptr, nullptr);
+
+  try
+  {
+    NVBENCH_CUDA_CALL_RESET_ERROR(cudaStreamSynchronize(0));
+    return 1;
```

Comment:
Add an assert here that always fails if the the NVBENCH_CUDA_CALL macro doesn't throw.

Reply:
Done.